pkg/tagger/utils/expiry.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package utils
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+// Expire regularly pools the source keeping the elements reporting.
```

Comment:
Expire does not pool a source but has the info pushed to it via `Update`. I'd write:

> Expire implements a simple last-seen-time-based expiry logic for watching for disappearing entities (for triggering events or just cache housekeeping).
> User classes define an expiry delay, then call `Update` every time they encounter a given entity. `ComputeExpires()` returns the entity names that have not been seen for longer than the configured delay.
> As Expire keeps an internal state of entity names, `Update` will return true if a name is new, false otherwise.

Reply:
👌